src/NTokenGeometricMean/NTokenGeometricMeanMath.sol

```diff
@@ -19,7 +19,7 @@ function computeTradingFunction(
         uint256 a = uint256(
             int256(reserves[i].divWadDown(L)).powWad(int256(params.weights[i]))
         );
```

Comment:
Would recommend rounding up `reserves[i].divWadDown(L)` here as well to be consistent with `G3MMath.computeTradingFunction()`:

[G3MMath.sol#L18-L19](https://github.com/primitivefinance/DFMM/blob/fd867a7bd8a28f56e0c7560244e5c195ab12f577/src/GeometricMean/G3MMath.sol#L18-L19)

```solidity
    uint256 a = uint256(int256(rX.divWadUp(L)).powWad(int256(params.wX)));
    uint256 b = uint256(int256(rY.divWadUp(L)).powWad(int256(params.wY)));
```

**Recommendation:**

```diff
  uint256 a = uint256(
-     int256(reserves[i].divWadDown(L)).powWad(int256(params.weights[i]))
+     int256(reserves[i].divWadUp(L)).powWad(int256(params.weights[i]))
  );
```

Reply:
Needs to be fixed still @clemlak